### PR TITLE
Create and Update functionality for gardens

### LINKED ISSUE #30: Functionality to Create Gardens
User Story 11, Parent Creation (x2)

As a visitor
When I visit the Parent Index page
Then I see a link to create a new Parent record, "New Parent"
When I click this link
Then I am taken to '/parents/new' where I  see a form for a new parent record
When I fill out the form with a new parent's attributes:
And I click the button "Create Parent" to submit the form
Then a `POST` request is sent to the '/parents' route,
a new parent record is created,
and I am redirected to the Parent Index page where I see the new Parent displayed.

### LINKED ISSUE #31: Functionality to Update Gardens
User Story 12, Parent Update (x2)

As a visitor
When I visit a parent show page
Then I see a link to update the parent "Update Parent"
When I click the link "Update Parent"
Then I am taken to '/parents/:id/edit' where I  see a form to edit the parent's attributes:
When I fill out the form with updated information
And I click the button to submit the form
Then a `PATCH` request is sent to '/parents/:id',
the parent's info is updated,
and I am redirected to the Parent's Show page where I see the parent's updated info

### PR DESCRIPTION
**Pull request by mkrumholz**
## Changes

- Added a create and update form for gardens
- Linked those forms on the index
- Added a link to the garden show page for updating
- Added the `bootstrap_form` gem for styling the forms

## Special notes

I was going to ask Brian tomorrow about the bootstrap_form gem, but I assume it's OK since it's just for styling. You can do whatever with that, ignore it or use it to style your forms too. It's pretty usable and there's great documentation here: https://github.com/bootstrap-ruby/bootstrap_form

That said, if you do everything as normal, this will just not affect pages where you don't explicitly call it.

## Checklist

- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%

closes #30 
closes #31

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,6 +11,8 @@ gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 3.11'
 # Use SCSS for stylesheets
 gem 'sass-rails', '~> 5.0'
+# Use Bootstrap forms gem for form formatting
+gem "bootstrap_form", "~> 4.0"
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
```
Comment (Jaybraum) [inline Gemfile:18]: There are some Gems here that I don't recognise, lets go over during the next check in.

File: spec/features/gardens/edit_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# As a visitor
+# When I visit a parent show page
+# Then I see a link to update the parent "Update Parent"
+# When I click the link "Update Parent"
+# Then I am taken to '/parents/:id/edit' where I see a form to edit the parent's attributes:
+# When I fill out the form with updated information
+# And I click the button to submit the form
+# Then a PATCH request is sent to '/parents/:id',
+# the parent's info is updated,
+# and I am redirected to the Parent's Show page where I see the parent's updated info
```
Comment (Jaybraum) [inline spec/features/gardens/edit_spec.rb:10]: I don’t know if you left the user story here as reference? Can we leave them in?

**Comment by Jaybraum** [comment]
Seems all good to me, I don't know if these conflicts are from the added file form the last merge

**Comment by mkrumholz** [comment]
Just reopening here.

**Pull request merged**